torch/utils/backend_registration.py

```diff
@@ -292,3 +297,38 @@ def generate_methods_for_privateuse1_backend(for_tensor: bool = True, for_module
 
     if for_storage:
         _generate_storage_methods_for_privateuse1_backend(custom_backend_name, unsupported_dtype)
+
+def _get_custom_mod_func(func_name: str):
+    r"""
+    Return the func named `func_name` defined in custom device module. If not defined,
+    return `None`. And the func is registered with `torch.utils.rename_privateuse1_backend('foo')`
+    and `torch._register_device_module('foo', BackendModule)`.
+    If the custom device module or the func is not defined, it will give warning or error message.
+    Args:
+        func_name (str): return the callable func named func_name defined in custom device module.
+    Example::
+        class DummyfooModule:
+            @staticmethod
+            def is_available():
+                return True
+            @staticmethod
+            def func_name(*args, **kwargs):
+                ....
+        torch.utils.rename_privateuse1_backend("foo")
+        torch._register_device_module("foo", DummyfooModule)
+        foo_is_available_func = torch.utils.backend_registration._get_custom_mod_func("is_available")
+        if foo_is_available_func:
+            foo_is_available = foo_is_available_func()
+        func_ = torch.utils.backend_registration._get_custom_mod_func("func_name")
+        if func_:
+            result = func_(*args, **kwargs)
+    Attention: This func is only used for custom device.
+    """
+    assert isinstance(func_name, str), f"func_name must be `str`, but got `{type(func_name)}`."
+    custom_device_mod = getattr(torch, _privateuse1_backend_name, None)  # type: ignore[arg-type]
+    if custom_device_mod is None:
+        message = f'Try to use torch.{_privateuse1_backend_name}.{func_name}. The backend must register a custom backend '
+        message += f"module with `torch._register_device_module('{_privateuse1_backend_name}', BackendModule)`. And "
+        message += f"BackendModule needs to have the following API's:\n `{func_name}(*args, **kwargs)`. \n"
+        warnings.warn(message)
+    return getattr(custom_device_mod, func_name, None)
```

Comment:
Instead of warning and returning `None`, can we just raise an error here? We'd probably rather fail fast if someone is missing this API (by trying to use it without having set a backend module beforehand).

Reply:
> Instead of warning and returning `None`, can we just raise an error here? We'd probably rather fail fast if someone is missing this API (by trying to use it without having set a backend module beforehand). If you do then you can also remove the warning (and just raise an exception).

yes, I have change it to raise an runtimeerror.